k8s/provider.go

```diff
@@ -20,6 +21,8 @@ const (
 	podNameEnvVar                   = "KUBERNETES_POD_NAME"
 	consulPodNameLabelTemplate      = "k8sPodName: %s"
 	consulPodNamespaceLabelTemplate = "k8sPodNamespace: %s"
+
+	getIPTimeoutSeconds = 10
```

Comment:
Please make this a default value, but allow to configure this by environment value at least.